### ISSUE TITLE
🐛 Linting rule should not trigger multiple times for the same line

### ISSUE THREAD

**Issue by mroy-seedbox** [codingjoe/relint#114]
### Bug Description

Sometimes a rule can capture the same match multiple times.

### Expected Behavior

That should be counted as just one match.

### Version

3.3.1

### Steps to Reproduce

```yml
- name: Jinja expressions should have a leading and trailing space
  pattern: '([{][{][^ ])|([^ ][}][}])'
  hint: "Add spaces before and after the curly braces: {{ code here }}"
```

And lint: `{{"hello world"}}`
